aiohomekit/model/__init__.py

```diff
@@ -407,16 +407,29 @@ def has_aid(self, aid: int) -> bool:
         """Return True if the given aid exists."""
         return aid in self._aid_to_accessory
 
-    def process_changes(self, changes: dict[tuple[int, int], dict[str, Any]]) -> None:
-        """Process changes from a HomeKit controller."""
-        for (aid, iid), value in changes.items():
+    def process_changes(
+        self, changes: dict[tuple[int, int], dict[str, Any]]
+    ) -> set[tuple[int, int]]:
+        """Process changes from a HomeKit controller.
+
+        Returns a set of the changes that were applied.
+        """
+        changed: set[tuple[int, int]] = set()
+        for aid_iid, value in changes.items():
+            (aid, iid) = aid_iid
             if not (char := self.aid(aid).characteristics.iid(iid)):
                 continue
 
```

Comment:
Should we just always add it to changed if it's one of the event characteristics?

Reply:
Yes we will need to do that

Reply:
https://github.com/Jc2k/aiohomekit/pull/339/files#diff-5443a6f050cba5fd14b7604e81e8139f66401a7f2eb74c8e8f396837127b2b3cR425